src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1954,24 +1954,31 @@ def create_artifact_version(
         """
         with Session(self.engine) as session:
             # Check if an artifact with the given name and version exists
-            existing_artifact = session.exec(
-                select(ArtifactVersionSchema)
-                .where(
-                    ArtifactVersionSchema.artifact_id
-                    == artifact_version.artifact_id
-                )
-                .where(
-                    ArtifactVersionSchema.version == artifact_version.version
-                )
-            ).first()
-            if existing_artifact is not None:
-                raise EntityExistsError(
-                    f"Unable to create artifact with name "
-                    f"'{existing_artifact.artifact.name}' and version "
-                    f"'{artifact_version.version}': An artifact with the same "
-                    "name and version already exists."
+            def _check(tolerance: int = 0) -> None:
```

Comment:
why would this ever be >0? or why would we want to make this >0?

Reply:
If I check it initially, I expect no version, but right before committing the transaction, I want to be sure that there is exactly 1 version, while it could be 2 if someone committed between my actions in this method. Thus having a flexible tolerance factor.